### ISSUE TITLE
Cache removalListener Half done

### ISSUE THREAD

**Issue by rongflag** [google/guava#3346]
// 创建一个 cache类
	public static LoadingCache<String, Object> cache = CacheBuilder.newBuilder()
			// 最大的内存个数
			.maximumSize(1000).recordStats()
			// 写之后 10s 删除
			.expireAfterWrite(10, TimeUnit.SECONDS)
			// 移除监听器。可以用来做机台的关机。比如当几分钟没有上传数据 就 当关机
			.removalListener(new RemovalListener<String, Object>() {

				@Override
				public void onRemoval(RemovalNotification<String, Object> notification) {
					System.out.println("remove=>" + notification.getKey() + ":" + notification.getValue());
				}
			}).build(
					// 获取策略 请求拿值的时候没有这个key 做什么操作和返回什么
					new CacheLoader<String, Object>() {
						@Override
						public Object load(String key) throws Exception {
							System.out.println("no load=>" + System.currentTimeMillis());
							return key;
						}

					});

	public static void main(String[] args) throws ExecutionException {
		for (int i = 0; i < 5; i++) {
			Gcache.put("" + i, i);
		}

		while (true) {
			Object ifPresent = Gcache.getIfPresent("" + 1);
			if (ifPresent == null) {
				System.out.println(System.currentTimeMillis());
				break;
			} else {
				System.out.println(System.currentTimeMillis() + "====" + ifPresent);
			}
			try {
				Thread.sleep(2000);
			} catch (InterruptedException e) {
				e.printStackTrace();
			}
		}
		Object ifPresent = Gcache.get("" + 5);
	}

1545294260626====1
1545294262627====1
1545294264628====1
1545294266629====1
1545294268629====1
1545294270641
remove=>0:0
remove=>1:1
no load=>1545294270662
this is my prints. so Random entry  “onRemoval”? I want to do somethings when it in "onRemoval" version - 27.0-jre

**Comment by ben-manes** [comment]
Guava's cache does not create any threads, e.g. to schedule an expiration. Instead it waits until activity on the cache occurs, e.g. reads, and performs maintenance cycles. You can call `cleanUp()` to trigger this work.

**Comment by ben-manes** [comment]
This feature, of using an optional scheduling thread, is released in Caffeine 2.8.

```java
Caffeine.newBuilder()
    .scheduler(Scheduler.systemScheduler()) // Java 9+
    .expireAfterWrite(10, TimeUnit.SECONDS)
    .removalListener(...)
    .build();
```

If you are not on Java 9+ then you will have to supply your own thread, e.g. using a `Scheduler.forScheduledExecutorService(se)`.
